plugins/filter_aws/aws.c

```diff
@@ -76,6 +93,18 @@ static int cb_aws_init(struct flb_filter_instance *f_ins,
     ctx->availability_zone_len = 0;
     ctx->instance_id = NULL;
     ctx->instance_id_len = 0;
+    ctx->instance_type = NULL;
+    ctx->instance_type_len = 0;
+    ctx->private_ip = NULL;
+    ctx->private_ip_len = 0;
+    ctx->vpc_id = NULL;
+    ctx->vpc_id_len = 0;
+    ctx->ami_id = NULL;
+    ctx->ami_id_len = 0;
+    ctx->account_id = NULL;
+    ctx->account_id_len = 0;
+    ctx->hostname = NULL;
+    ctx->hostname_len = 0;
```

Comment:
This plugin was some of the first C code I ever wrote... so it has some unneeded things in it. All of this (everything under the "initilize all fields" comment is not needed. When the `ctx` is created with flb_calloc it will zero the memory. (As opposed to malloc, which does not initialize the memory it allocates).

Reply:
Good to know! Removed all these initial code.